mlflow/server/auth/__init__.py

```diff
@@ -910,9 +931,17 @@ def create_app(app: Flask = app):
     _logger.warning(
         "This feature is still experimental and may change in a future release without warning"
     )
-    # secret key required for flashing
-    if not app.secret_key:
-        app.secret_key = str(uuid.uuid4())
+
+    # a secret key is required for flashing, and also for
+    # CSRF protection. it's important that this is a static key,
+    # otherwise CSRF validation won't work across workers.
+    app.secret_key = os.environ.get(FLASK_SERVER_SECRET_KEY, str(uuid.uuid4()))
```

Comment:
Sorry for the basic question, but do we support running multiple mlflow tracking servers behind a shared load balancer?

Reply:
i'm not sure how common that is, though i guess in theory it's possible. i'm not sure if it's worth optimizing for this use-case though, as it would add a new requirement to run the auth app (currently users don't have to specify anything manually)

Reply:
But the CSRF verification does not work for the multi-server architecture with the current code, correct? (Sorry, I previously misunderstood the configuration of FLASK_SERVER_SECRET_KEY part. That's actually set by the command, not by users). Then, I think we should at least mention it somewhere if it is difficult to support inputting a secret key, since supporting horizontal scaling is a common pattern for servers with CSRF protection.